internal/http/services/owncloud/ocdav/copy.go

```diff
@@ -52,18 +52,34 @@ func (s *svc) handlePathCopy(w http.ResponseWriter, r *http.Request, ns string)
 	ctx, span := rtrace.Provider.Tracer("reva").Start(r.Context(), "copy")
 	defer span.End()
 
+	// If HTTP third-party copy mode is enabled, go for that
+	if s.c.EnableHTTPTpc {
+
+		if r.Header.Get("Source") == "" {
```

Comment:
Shouldn't both local as well as third party copies be supported when TPC is enabled? We can carry out checks based on the headers.
- If source != "" -> TPC pull
- If dst is a proper URL -> TPC push
- Else if dst != "" -> local copy
- Else return error